### PR TITLE
added password hashing

### PR DESCRIPTION
**Pull request by Martynice**

### CODE REVIEW

**Review by OleksiiTromsa — APPROVED** [review]
My decision is almost the same, so I have nothing to add)

File: src/main/java/com/internet/shop/model/User.java [hunk 1]
```diff
@@ -22,6 +23,21 @@ public User(Long id, String name, String login, String password) {
...
     }
 
+    public User(String name, String login, String password, byte[] salt) {
+        this.name = name;
+        this.login = login;
+        this.password = password;
+        this.salt = salt;
+    }
+
+    public User(Long id, String name, String login, String password, byte[] salt) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/model/User.java:19]: there are too many constructors at the moment, do we need them all?

File: src/main/java/com/internet/shop/util/HashUtil.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            }
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("Couldn't hash the password", e);
+        }
+        return hashedPassword.toString();
+    }
+
+    public static boolean isValid(String password, User user) {
+        return hashPassword(password, user.getSalt()).equals(user.getPassword());
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/util/HashUtil.java]: let's make this method private in AuthenticationServiceImpl class

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/internet/shop/model/User.java (hunk 1 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/model/User.java:19]: removed

File: src/main/java/com/internet/shop/util/HashUtil.java (hunk 2 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/util/HashUtil.java]: replaced

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -16,9 +16,13 @@
...
     public User login(String login, String password) throws AuthenticationException {
         Optional<User> userFromDB = userService.findByLogin(login);
-        if (userFromDB.isEmpty() || !HashUtil.isValid(password, userFromDB.get())) {
+        if (userFromDB.isEmpty() || !isValid(password, userFromDB.get())) {
             throw new AuthenticationException("Incorrect login or password");
         }
         return userFromDB.get();
     }
+
+    private static boolean isValid(String password, User user) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]:
```suggestion
    private boolean isValid(String password, User user) {
```

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**